rollover.py

```diff
@@ -0,0 +1,300 @@
+    parser.add_argument('-s',
+                        '--scale-down',
+                        action="store_true",
+                        help="scale the cluster down instead of rolling over")
+    parser.add_argument('-t',
+                        '--timeout',
+                        type=int,
+                        default=30,
+                        help="`docker stop` timeout")
+    parser.add_argument('-n',
+                        '--noop',
+                        action="store_true",
+                        help="dry run. Don't actually make changes.")
+    parser.add_argument('cluster',
+                        help="fully qualified name of the cluster")
+    parser.add_argument('asg',
+                        help="auto scaling group for the cluster")
+    args = parser.parse_args()
+
+    #
+    # Create AWS connections
+    #
+    ecs_client = ecs.ECSClient(args.cluster)
+    ec2_client = ec2.EC2Client()
+    asg = scaling.AutoScalingGroup(args.asg)
+
+    # Prompt the user for the instances to adjust
+    cluster_instances = ecs_client.list_cluster_instances()
+    selected_instances = prompt_for_instances(cluster_instances)
+    if not selected_instances:
+        return
+
+    #
+    # learn about and sort the instances by availability zone
+    #
+    ecs_descriptions = ecs_client.describe_instances(selected_instances)
+    ec2_to_ecs_id = dict([(v['ec2InstanceId'], k) for k, v in ecs_descriptions.items()])
+    ec2_ids = ec2_to_ecs_id.keys()
+    ec2_descriptions = ec2_client.describe_instances(ec2_ids)
+
+    # sort and warn by availability zones
+    asg_instances = asg.describe_instances()
+    sorted_ec2_ids = sort_and_check_availability_zones(ec2_descriptions,
+                                                       asg_instances)
+    if not sorted_ec2_ids:
+        return
+
+    #
+    # Query all services and tasks on each ECS instance
+    #
+    sys.stdout.write("Querying ECS services...")
+    sys.stdout.flush()
+    service_ids = ecs_client.list_services()
+    service_descriptions = ecs_client.describe_services(service_ids)
+    service_events = map_service_events(service_descriptions)
+
+    task_ids = ecs_client.list_tasks()
+    task_descriptions = ecs_client.describe_tasks(task_ids)
+    ecs_instance_services = map_instance_services(service_descriptions,
+                                                  task_descriptions)
+    asg_ids = [asg_instance['InstanceId'] for asg_instance in asg_instances]
+    sorted_asg_ids = [ec2_id for ec2_id in sorted_ec2_ids if ec2_id in asg_ids]
+    print "done"
+
+    #
+    # Iterate through each instance
+    #
+    for ec2_id in sorted_asg_ids:
+        ecs_id = ec2_to_ecs_id[ec2_id]
+        print "Preparing to remove %s (%s)" % (ecs_id, ec2_id)
+
+        #
+        # Remove ECS instance from scaling group
+        #
+        sys.stdout.write("Removing EC2 instance %s from scaling group and waiting for replacements..." % (ec2_id))
```

Comment:
This comment should probably differ if scale_down was passed -- there wouldn't be any "waiting for replacements"